src/wallet/wallet.cpp

```diff
@@ -4509,20 +4533,32 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(const std::string& walle
     std::set<fs::path> wallet_dirs;
     if (success) {
         // Migration successful, unload all wallets locally, then reload them.
-        // Reload the main wallet
         wallet_dirs.insert(fs::PathFromString(local_wallet->GetDatabase().Filename()).parent_path());
-        success = reload_wallet(local_wallet);
-        res.wallet = local_wallet;
-        res.wallet_name = wallet_name;
+        // Reload and set the main spendable wallet only if needed
+        if (!empty_local_wallet) {
+            success = reload_wallet(local_wallet);
+            res.wallet = local_wallet;
+            res.wallet_name = wallet_name;
+        } else {
+            // At this point, the main wallet is empty after migration, meaning it has no records.
+            // Therefore, we can safely remove it.
+            fs::path path_to_remove = fs::PathFromString(local_wallet->GetDatabase().Filename()).parent_path();
```

Comment:
This also deletes the backup file.

Reply:
Upps, corrected. Thanks. Was interesting to know that we have no backup file existence general check. Added one now.